lib/house.rb

```diff
@@ -0,0 +1,35 @@
+class House
+
+  attr_reader :price, :address, :rooms
+  def initialize(price, address)
+    @price = price
+    @address = address
+    @rooms = []
+  end
+
+  def add_room(room)
+    @rooms << room
+  end
+
+  def rooms_from_category(room_category)
+    @rooms.find_all {|room| room.category == room_category}
+  end
+
+  def area
+    area = 0
+    @rooms.each {|room| area = area + room.area}
+    area
+  end
+
+  def house_details
+    details = {
+      "price" => price,
+      "address" => address
+    }
+    details
+  end
+
+  def price_per_square_foot
+    (price.delete_prefix("$").to_f / area).round(2)
```

Comment:
don't get too used to `delete_prefix`-- it's part of the very latest and greatest version of ruby, which your place of work will almost certainly not have 🙂  I'd try to find other ways to do this.